### PR TITLE
Pro: add referrers to dashboard and single article stats

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description

Adds referrers information for the pro dashboard and the article stats

Currently shows only the top 20 domains with the most referrals.

This PR adds two columns to the `page_views` table: `domain`  and `path`. I presume this table is massive, which it's not going to help the migration because `ALTER TABLE ADD COLUMN` rewrites the table and it will incur in downtime.

**Notes before merging**:

- the migration could be massive and slow
- there's a temporary Rake script (to be removed after deploy) that is going to backfill these new columns

I don't know how big the table is, the script can be sped up by only backfilling page views created after the stats release date (`2019-04-01`) or only for Pro users for now.

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

This box appears under the existing stats boxes:

Screenshot_2019-06-24 Pro Analytics - DEV(local) Community 👩‍💻👨‍💻

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

File: app/javascript/analytics/dashboard.js [hunk 1]
```diff
@@ -165,48 +165,65 @@ function drawCharts(data, timeRangeLabel) {
...
+    .map(referrer => {
+      return `
+      <tr>
+        <td>${referrer.domain}</td>
+        <td>${referrer.count}</td>
+      </tr>
+    `;
+    });
+
+  // add referrers with empty domains if present
```
Comment (rhymes) [inline app/javascript/analytics/dashboard.js]: Some page views don't have a referrer, hence they won't have a domain. We append this at the top of the table

File: lib/tasks/temporary/page_views.rake [hunk 2]
```diff
@@ -0,0 +1,17 @@
+namespace :page_views do
```
Comment (rhymes) [inline lib/tasks/temporary/page_views.rake]: Temporary script to be run after the migration, to be removed afterwards

**Comment by benhalpern** [comment]
These are famous last words but I _think_ adding a column without default values should be okay.

If we attempt this migration and it times out we'll be ready to roll back the deploy.

**Comment by rhymes** [comment]
@benhalpern ok I separated the add index statement to run concurrently in a separate migration, the backfill script has also been modified to log possible errors and to be retried in case. It's still going to be a long ordeal nonetheless :( 

I couldn't find an easy way to backfill using SQL, which would have been faster

**Pull request merged**